src/main/java/org/shanerx/tradeshop/commands/commandrunners/ShopUserCommand.java

```diff
@@ -0,0 +1,260 @@
+ *
+ */
+
+package org.shanerx.tradeshop.commands.commandrunners;
+
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.shanerx.tradeshop.TradeShop;
+import org.shanerx.tradeshop.commands.CommandPass;
+import org.shanerx.tradeshop.data.config.Message;
+import org.shanerx.tradeshop.data.config.Setting;
+import org.shanerx.tradeshop.data.config.Variable;
+import org.shanerx.tradeshop.framework.ShopChange;
+import org.shanerx.tradeshop.framework.events.PlayerShopChangeEvent;
+import org.shanerx.tradeshop.player.Permissions;
+import org.shanerx.tradeshop.player.ShopRole;
+import org.shanerx.tradeshop.player.ShopUser;
+import org.shanerx.tradeshop.shop.Shop;
+import org.shanerx.tradeshop.shoplocation.ShopLocation;
+import org.shanerx.tradeshop.utils.objects.ObjectHolder;
+import org.shanerx.tradeshop.utils.objects.Tuple;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+enum UserOperationStatus {
+    SUCCESSFUL(Message.UPDATED_SHOP_USERS_SUCCESSFUL),
+    FAILED(Message.UPDATED_SHOP_USERS_FAILED),
+    FAILED_CAPACITY(Message.UPDATED_SHOP_USERS_FAILED_CAPACITY),
+    FAILED_EXISTING(Message.UPDATED_SHOP_USERS_FAILED_EXISTING),
+    FAILED_MISSING(Message.UPDATED_SHOP_USERS_FAILED_MISSING);
+
+    private final Message text;
+
+    UserOperationStatus(Message text) {
+        this.text = text;
+    }
+
+    @Override
+    public String toString() {
+        return text.toString();
+    }
+}
+
+/**
+ * Implementation of CommandRunner for commands that view/change shop users
+ *
+ * @since 2.6.0
+ */
+public class ShopUserCommand extends CommandRunner {
+
+    private OfflinePlayer target;
+
+    public ShopUserCommand(TradeShop instance, CommandPass command) {
+        super(instance, command);
+    }
+
+    /**
+     * Tells the player who the Owner/Managers/Members that are on the shop are
+     */
+    public void who() {
+        String owner = "";
+        StringBuilder managers = new StringBuilder();
+        StringBuilder members = new StringBuilder();
+        Shop shop = findShop();
+
+        if (shop == null)
+            return;
+
+        if (shop.getShopType().isITrade()) {
+            Message.WHO_MESSAGE.sendMessage(pSender,
+                    new Tuple<>(Variable.OWNER.toString(), Setting.ITRADESHOP_OWNER.getString()),
+                    new Tuple<>(Variable.MANAGERS.toString(), "None"),
+                    new Tuple<>(Variable.MEMBERS.toString(), "None"));
+            return;
+        }
+
+        if (shop.getOwner() != null)
+            owner = shop.getOwner().getName();
+
+        if (shop.hasUsers(ShopRole.MANAGER)) {
+            for (ShopUser usr : shop.getUsers(ShopRole.MANAGER)) {
+                if (managers.toString().equals(""))
+                    managers = new StringBuilder(usr.getName());
+                else
+                    managers.append(", ").append(usr.getName());
+            }
+        }
```

Comment:
> Up to you though since I think it'll most like be a minor cleanup/reformat(can probably use a Collections.forEach or other lambda loop to accomplish this.

Yes this would be better I think. There would not be any performance benefit, it is just easier to read that way I  think

Reply:
I'm stupid, here's the rewrite. apparently i forgot I had a getUserNames method or I didn't updaate this after making that -_-

```java
public void who() {
        Shop shop = ShopUser.findObservedShop(command.getPlayerSender());

        if (shop == null)
            return;

        if (shop.getShopType().isITrade()) {
            Message.WHO_MESSAGE.sendMessage(command.getPlayerSender(),
                    new Tuple<>(Variable.OWNER.toString(), Setting.ITRADESHOP_OWNER.getString()),
                    new Tuple<>(Variable.MANAGERS.toString(), "None"),
                    new Tuple<>(Variable.MEMBERS.toString(), "None"));
            return;
        }

        String owner = String.join(", ", shop.getUserNames(ShopRole.OWNER));
        String managers = String.join(", ", shop.getUserNames(ShopRole.MANAGER));
        String members = String.join(", ", shop.getUserNames(ShopRole.MEMBER));
        
        Message.WHO_MESSAGE.sendMessage(command.getPlayerSender(),
                new Tuple<>(Variable.OWNER.toString(), owner.length() < 3 ? "None" : owner),
                new Tuple<>(Variable.MANAGERS.toString(), managers.length() < 3 ? "None" : managers),
                new Tuple<>(Variable.MEMBERS.toString(), members.length() < 3 ? "None" : members));
    }
```